src/search/pdbs/cegar.cc

```diff
@@ -521,78 +521,63 @@ void CEGAR::refine(const FlawList &flaws) {
-            if (verbosity >= utils::Verbosity::VERBOSE) {
-                utils::g_log << "merge the two patterns" << endl;
+            if (log.is_at_least_verbose()) {
+                log << "merge the two patterns" << endl;
             }
             merge_patterns(collection_index, other_index);
             added_var = true;
         }
     } else {
         // Variable is not yet in the collection.
-        if (verbosity >= utils::Verbosity::VERBOSE) {
-            utils::g_log << "var" << var
-                         << " is not in the collection yet" << endl;
+        if (log.is_at_least_verbose()) {
+            log << "var" << var
+                << " is not in the collection yet" << endl;
         }
         if (can_add_variable_to_pattern(collection_index, var)) {
-            if (verbosity >= utils::Verbosity::VERBOSE) {
-                utils::g_log << "add it to the pattern" << endl;
+            if (log.is_at_least_verbose()) {
+                log << "add it to the pattern" << endl;
             }
             add_variable_to_pattern(collection_index, var);
             added_var = true;
         }
     }
 
     if (!added_var) {
-        if (verbosity >= utils::Verbosity::VERBOSE) {
-            utils::g_log << "could not add var/merge pattern containing var "
-                         << "due to size limits, blacklisting var" << endl;
+        if (log.is_at_least_verbose()) {
+            log << "could not add var/merge pattern containing var "
+                << "due to size limits, blacklisting var" << endl;
         }
         blacklisted_variables.insert(var);
     }
 }
 
 PatternCollectionInformation CEGAR::compute_pattern_collection() {
-    if (verbosity >= utils::Verbosity::NORMAL) {
-        utils::g_log << "CEGAR options:" << endl;
-        utils::g_log << "max pdb size: " << max_pdb_size << endl;
-        utils::g_log << "max collection size: " << max_collection_size << endl;
-        utils::g_log << "max time: " << max_time << endl;
-        utils::g_log << "wildcard plans: " << use_wildcard_plans << endl;
-        utils::g_log << "verbosity: ";
-        switch (verbosity) {
-        case utils::Verbosity::SILENT:
-            utils::g_log << "silent";
-            break;
-        case utils::Verbosity::NORMAL:
-            utils::g_log << "normal";
-            break;
-        case utils::Verbosity::VERBOSE:
-            utils::g_log << "verbose";
-            break;
-        case utils::Verbosity::DEBUG:
-            utils::g_log << "debug";
-            break;
-        }
-        utils::g_log << endl;
-        utils::g_log << "goal variables: ";
+    if (log.is_at_least_normal()) {
+        log << "CEGAR options:" << endl;
+        log << "max pdb size: " << max_pdb_size << endl;
+        log << "max collection size: " << max_collection_size << endl;
+        log << "max time: " << max_time << endl;
+        log << "wildcard plans: " << use_wildcard_plans << endl;
+        log.dump_options();
```

Comment:
No need to change this, but I'm not sure if we actually want this here going forward. This might end up being a complex log configuration, possibly for a log shared with other components of the planner although we don't currently support that, and it's not clear to me whether we then want to print a complex log configuration in a place like this. Just something to potentially consider in the future as the log functionality evolves further.